app/models/user.rb

```diff
@@ -27,6 +27,10 @@ def to_s
     self.name
   end
 
+  def all_emails
+    [self.email] + self.additional_emails.map {|x| x.email}
```

Comment:
Redundant `self` detected.<br>Space between { and | missing.<br>Space missing inside }.<br>Pass `&:email` as an argument to `map` instead of a block.